src/components/Dialog.tsx

```diff
@@ -62,8 +62,7 @@ const DialogPortal = ({
   container?: HTMLElement;
   children: React.ReactNode;
 }) => {
-  const { dialogArea } = useDialogArea();
-
+  const dialogArea = useDialogArea()?.dialogAreaRef.current;
```

Comment:
Prefer destructuring here.

`const { dialogAreaRef } = useDialogArea() ?? {};` or if we return it directly `const { current } = useDialogArea() ?? {}`

Reply:
is this mostly a coding style preference (which is fine) or is there a deeper reasoning to always preferring destructuring? either way is fine, i'm not pressed about it

Reply:
I think you kind of answered in your prev comment :)

> was thinking about this, but i opted for returning in object for:
> 
> consistency with other hooks
> extendability. if we ever want to add functionality to this we'll have less code changes if we keep the return in obj format

Reply:
:) just keeping as many things in as flexible notation as possible, makes sense